app/javascript/listings/listings.jsx

```diff
@@ -284,27 +301,28 @@ export class Listings extends Component {
 
   listingSearch(query, tags, category, slug) {
     const t = this;
-    const { index, page, listings } = t.state;
-    const filterObject = { tagFilters: tags, hitsPerPage: 75, page };
-    if (category.length > 0) {
-      filterObject.filters = `category:${category}`;
-    }
-    index.search(query, filterObject).then(function searchDone(content) {
-      const fullListings = listings;
-      content.hits.forEach(listing => {
-        if (listing.bumped_at) {
-          if (!listings.map(l => l.id).includes(listing.id)) {
-            fullListings.push(listing);
-          }
-        }
-      });
+    const { page, listings } = t.state;
+    const dataHash = {
+      per_page: 75,
+      page,
+      classified_listing_search: query,
+    };
+
+    if (dataHash.tags) dataHash.tags = tags;
```

Comment:
After #6412 was merged, we removed some backend checks for empty params. At this stage, `tags` is often an empty string so I made this a conditional to avoid sending `tags=""` to search which ends up returning nothing when if we do let that through.

Reply:
Since we're not allowing empty param searches for ClassifiedListings specifically, I think we should document that here in the frontend code for this method in particular. If (or when?) we decide to disallow empty param searches for _all_ search endpoints, we can easily figure out which ones have accounted for this or not, and then abstract it into `fetchSearch`.

Another thing I'm curious about: why _not_ just disallow all empty param searches outright? Is there a reason (or benefit) to only make this change to classified listings? Do we ever actually want to allow users to search with empty params and hit the endpoint?